curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-pyramid-generator/660f0a55847d6cc485f29ba5.md

```diff
@@ -0,0 +1,52 @@
+---
+id: 660f0a55847d6cc485f29ba5
+title: Step 21
+challengeType: 0
+dashedName: step-21
+---
+
+# --description--
+
+A <dfn>method</dfn> in JavaScript is a function that's associated with certain values or objects. An example you've already encountered is the `.log()` method, which is part of the `console` object.
+
+Arrays have their own methods, and the first you will explore is the `.push()` method. This allows you to "push" a value to the end of an array. Here is an example to add the number `12` to the end of an array:
+
+```js
+array.push(12);
+```
+
+Use `.push()` to add the string `"freeCodeCamp"` to the end of your `rows` array. Add this code before your `console.log` so you can see the change you made to your array.
+
+# --hints--
+
+You should use `.push()` in your code.
+
+```js
+assert.match(code, /\.push\(/);
+```
+
+You should use the `.push()` method of your `rows` array.
+
+```js
+assert.match(code, /rows\.push\(/);
+```
+
+You should pass the string `"freeCodeCamp"` to your `.push()` method. Don't forget your semi-colon.
+
+```js
+assert.match(code, /rows\.push\(\s*('|")freeCodeCamp\1\s*\);/)
+```
```

Comment:
semi colons should be optional

````suggestion
You should pass the string `"freeCodeCamp"` to your `.push()` method. 

```js
assert.match(code, /rows\.push\(\s*('|")freeCodeCamp\1\s*\);?/)
```
````